insurance_cloudevents/README.md

```diff
@@ -0,0 +1,90 @@
+       - **TOKEN** = {your dkube authentication token}
+       - **DKUBE_URL** = {your dkube url}
+5. Run all the cells. This will create all the dkube resources required for this example automatically.
+
+### Launch IDE (Outside Dkube)
+
+#### Note: Follow the instructions if you are running Notebook IDE outside DKube, for example in VSCode. This is the case with minimal DKube. 
+
+1. Download [resources.ipynb](https://github.com/oneconvergence/dkube-examples/tree/monitoring/insurance/resources.ipynb)
+2. Open the notebook and fill the details in the first cell.
+   - **DKUBEUSERNAME** = {your dkube username}
+   - **MODELMONITOR_NAME** = {your model monitor name}
+   - **TOKEN** = {your dkube authentication token}
+   - **DKUBE_URL** = {your dkube url}
+   - **S3_KEY** = {MINIO access key}
+     - **S3_SECRET_KEY** = {MINIO access secret key}
+     - **S3_ENDPOINT** = {MINIO Endpoint}
+3. Run all the cells.
+
+## Section 2: Insurance Model Training
+
+#### Note: This uses DKube Runs, Kubeflow Pipelines and KfServing. It requires full Dkube installed. 
+
+1. From **workspace/insurance/insurance_cloudevents** open **train.ipynb** to build the pipeline.
+2. The pipeline includes preprocessing, training and serving stages. Run all cells
+     - **preprocessing**: the preprocessing stage generates the dataset (either training-data or retraining-data) depending on user choice.
+     - **training**: the training stage takes the generated dataset as input, train a sgd model and outputs the model.
+     - **serving**: The serving stage takes the generated model and serve it with a predict endpoint for inference.
+3. Verify that the pipeline has created the following resources
+     - Datasets: 'insurance-training-data' with version v2.
+     - Model: 'insurance-model' with version v2
+
+## Section 3: Modelmonitoring
+DKube provides Python SDK for creating a modelmonitor programmatically. You could also choose to create a modelmonitor from the DKube UI. This example cuurently support creating monitor using SDK. 
+
+- **Create Modelmonitor using SDK**
+1. From **workspace/insurance/insurance_cloudevents** open [modelmonitor.ipynb](https://github.com/oneconvergence/dkube-examples/tree/monitoring/insurance_cloudevents/modelmonitor.ipynb) and run all the cells. New model monitor will be created.
+2. Predict and Groundtruth dataset data will be generated by Data Generation step and will be utilised by modelmonitor.
+
+## Section 4: Data Generation
+1. Open [data_generation.ipynb](https://github.com/oneconvergence/dkube-examples/tree/monitoring/insurance_cloudevents/data_generation.ipynb) notebook for making predictions with the deployemnt endpoint and generate groundtruth datasets.
+2. In 1st cell, Update Frequency according to what you set in Modelmonitor for Drift. For eg: for 5 minutes, specify it as `5m` and to specify the frequency in hours use `5h` for 5 hours interval.
```

Comment:
Where is the labelled data stored? We should not require AWS to run this.

Unfortunately, we don't have append option to our datasets.  local dataset option in the insurance example only works with full dkube.

Reply:
We can keep local dataset option only for full dkube for now

Reply:
The predict data i.e cloudevents are being stored in {deployemnt-id}/YYYY/MM/DD/HH and the labelled data will be get stored in {deployemnt-id}/label_data in the same modelmonitor bucker.